src/jit/morph.cpp

```diff
@@ -18519,6 +18521,280 @@ void Compiler::fgAddFieldSeqForZeroOffset(GenTree* op1, FieldSeqNode* fieldSeq)
+                TopValue(1).Address(TopValue(0));
+                PopValue();
+                break;
+
+            case GT_FIELD:
+                if (node->AsField()->gtFldObj != nullptr)
+                {
+                    assert(TopValue(1).Node() == node);
+                    TopValue(1).Field(TopValue(0), node->AsField()->gtFldOffset);
+                    PopValue();
+                }
+                else
+                {
+                    assert(TopValue(0).Node() == node);
+                }
+                break;
+
+            case GT_OBJ:
+            case GT_BLK:
+            case GT_IND:
+                assert(TopValue(1).Node() == node);
+                TopValue(1).Indir(TopValue(0));
+                PopValue();
+                break;
+
+            case GT_CALL:
+                for (; TopValue(0).Node() != node; PopValue())
+                {
+                    if (TopValue(0).IsAddress())
+                    {
+                        bool fieldOnly = (TopValue(0).Node() == node->AsCall()->gtCallObjp);
+                        MarkAddressExposed(TopValue(0), fieldOnly);
+                    }
+                }
+                break;
+
+            default:
+                for (; TopValue(0).Node() != node; PopValue())
+                {
+                    if (TopValue(0).IsAddress())
+                    {
+                        MarkAddressExposed(TopValue(0));
+                    }
+                }
+                break;
+        }
+
+        assert(TopValue(0).Node() == node);
+        return Compiler::WALK_CONTINUE;
+    }
+
+private:
+    void MarkAddressExposed(const Value& val, bool fieldOnly = false)
+    {
+        assert(val.IsAddress());
+
+        unsigned   lclNum = val.GetLclNum();
+        LclVarDsc* varDsc = &m_compiler->lvaTable[lclNum];
+
+        m_compiler->lvaSetVarAddrExposed(lclNum);
+
+        if (varDsc->lvIsStructField)
+        {
+            if (!fieldOnly)
+            {
+                m_compiler->lvaSetVarAddrExposed(varDsc->lvParentLcl);
+            }
+        }
+#ifdef _TARGET_64BIT_
+        else if (!varDsc->lvIsParam && (genActualType(varDsc->TypeGet()) == TYP_INT))
+        {
+            if (Compiler::gtHasCallOnStack(&m_ancestors))
+            {
+                varDsc->lvQuirkToLong = true;
+            }
+        }
+#endif
+    }
+
+    void PushValue(GenTree* node)
+    {
+        m_valueStack.Push(node);
+    }
+
+    Value& TopValue(unsigned index)
+    {
+        return m_valueStack.IndexRef(index);
+    }
+
+    void PopValue()
+    {
+        m_valueStack.Pop();
+    }
+
+    void MorphStructField(GenTree* node, GenTree* user)
+    {
+        m_compiler->fgMorphStructField(node, user);
+    }
+
+    void MorphLocalField(GenTree* node, GenTree* user)
+    {
+        m_compiler->fgMorphLocalField(node, user);
+    }
+
+    bool FitsInOrNotLoc(GenTree* node, unsigned width)
+    {
+        return m_compiler->fgFitsInOrNotLoc(node, width);
```

Comment:
I do not like this `fgFitsInOrNotLoc`, I prefer  something like `getSize(node) <= width`. It will allow us to control cases where `fgFitsInOrNotLoc` returns false because it doesn't have enough information (for example when we have 'GT_ASG` to a `lcl_var`).

Reply:
Yup, this is just temporary to keep diffs low.

Reply:
Actually it wasn't even used.